### PR TITLE
add:nacos client

### PR DESCRIPTION
**Pull request by zhaoyunxing92**
* upadd:添加nacos-sdk-go
* update: 更新nacos-skd-go version [v1.0.8](https://github.com/nacos-group/nacos-sdk-go/releases/tag/v1.0.8)

**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: database/kv/nacos/client.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	sync.Mutex
+	namingClient map[string]naming_client.INamingClient
+}
+
+func initNacosClientPool() {
+	clientPool.namingClient = make(map[string]naming_client.INamingClient)
+}
+
+// NewNamingClient create nacos client
+func NewNamingClient(name string, share bool, sc []constant.ServerConfig, cc constant.ClientConfig) (naming_client.INamingClient, error) {
```
Comment (AlexStocks) [inline database/kv/nacos/client.go:45]: so long. pls split it.

File: database/kv/nacos/client.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	if share {
+		clientPoolOnce.Do(initNacosClientPool)
+		clientPool.Lock()
+		defer clientPool.Unlock()
+		if client, ok := clientPool.namingClient[name]; ok {
+			return client, nil
+		}
```
Comment (AlexStocks) [inline database/kv/nacos/client.go:52]: func() {
clientPool.Lock()		
defer clientPool.Unlock()		
if client, ok := clientPool.namingClient[name]; ok {			
 return client, nil		
}
}()

**Review by AlexStocks — APPROVED** [review]

File: database/kv/nacos/client.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	configMap := make(map[string]interface{}, 2)
+	configMap["serverConfigs"] = sc
+	configMap["clientConfig"] = cc
+	client, err := clients.CreateNamingClient(configMap)
+	if share {
```
Comment (gaoxinge) [inline database/kv/nacos/client.go:59]: Handle error before `if share`.

File: database/kv/nacos/client.go (hunk 2 above)
Comment (wenxuwan) [reply database/kv/nacos/client.go:53]: > func() {
> clientPool.Lock()
> defer clientPool.Unlock()
> if client, ok := clientPool.namingClient[name]; ok {
> return client, nil
> }
> }()

here should lock until NewNamingClient function finished, so can't put the lock in an func()

File: database/kv/nacos/client.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	sync.Mutex
+	namingClient map[string]naming_client.INamingClient
+}
+
+func initNacosClientPool() {
+	clientPool.namingClient = make(map[string]naming_client.INamingClient)
+}
+
+// NewNamingClient create nacos client
+func NewNamingClient(name string, share bool, sc []constant.ServerConfig,
```
Comment (wenxuwan) [inline database/kv/nacos/client.go:45]: Why not use name with database name, eg:NewNacosClient

Comment (zhaoyunxing92) [reply database/kv/nacos/client.go:45]: 这个是参考nacos那边的命名

**Review by LaurenceLiZhixin — APPROVED** [review]

**Comment by AlexStocks** [comment]
LGTM

File: database/kv/nacos/client.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+			return client, nil
+		}
+	}
+	cfg := vo.NacosClientParam{ClientConfig: &cc, ServerConfigs: sc}
+	client, err := clients.NewConfigClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	if share {
+		configClientPool.configClient[name] = client
```
Comment (georgehao) [inline database/kv/nacos/client.go:97]: here, maybe need a lock

**Review by georgehao — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

File: database/kv/nacos/client.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/nacos-group/nacos-sdk-go/clients/config_client"
+	"github.com/nacos-group/nacos-sdk-go/clients/naming_client"
+	"github.com/nacos-group/nacos-sdk-go/common/constant"
+	"github.com/nacos-group/nacos-sdk-go/vo"
+)
+
+var (
+	clientPool           nacosClientPool
+	configClientPool     nacosConfigClientPool
+	clientPoolOnce       sync.Once
```
Comment (AlexStocks) [inline database/kv/nacos/client.go:35]: 把 once 和 它保护的对象放在一块。把 once 变量挪动下位置。

File: database/kv/nacos/client.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	defer clientPool.Unlock()
+	if client, ok := clientPool.namingClient[name]; ok {
+		return client, nil
+	}
+
+	client, err := newNamingClient(sc, cc)
+	if err != nil {
+		return nil, err
+	}
+	clientPool.namingClient[name] = client
```
Comment (AlexStocks) [inline database/kv/nacos/client.go:74]: client, err := newNamingClient(sc, cc)	
if err == nil {		
 clientPool.namingClient[name] = client	
}

File: database/kv/nacos/client.go [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+	defer configClientPool.Unlock()
+	if client, ok := configClientPool.configClient[name]; ok {
+		return client, nil
+	}
+
+	client, err := newConfigClient(sc, cc)
+	if err != nil {
+		return nil, err
+	}
+	configClientPool.configClient[name] = client
```
Comment (AlexStocks) [inline database/kv/nacos/client.go:95]: client, err := newConfigClient(sc, cc)	
         if err == nil {	
             configClientPool.configClient[name] = client
         }

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**